table/block.h

```diff
@@ -387,6 +375,7 @@ class BlockIter final : public InternalIterator {
   std::vector<CachedPrevEntry> prev_entries_;
   int32_t prev_entries_idx_ = -1;
 
+ public:
```

Comment:
Can we leave the two Compare() functions virtual and implement differently in two child classes? This is to assume you want to reuse the binary search logic. Totally duplicating the binary search logic is also worth considering, as there may be some if/else there too.

Reply:
This refactoring is risk free. Making it virtual means 2 days to 1 week of extensive benchmarking to make sure that it does not have any side-effect. I suggest leaving such riskier changes for future PRs.